### ISSUE TITLE
heatmap of interpolation (i.e., temperature)

### ISSUE THREAD

**Issue by stevencollinscn** [ResidentMario/geoplot#63]
I'm so glad to see a package that makes plotting geo heatmap this kind of easy and beautiful. After reading the API and I realize that the heatmaps produced by geoplot using seaborn represent the density of geo-events. I wonder if there is any way you can think of to use interpolation method to generate those kinds of heatmaps representing air pollution, temperature, etc.

I've been looking for a way to do this kind of plot for [a white](https://gis.stackexchange.com/questions/298268/performing-interpolation-to-create-heatmap-in-python).

**Comment by ResidentMario** [comment]
Yes, `geoplot.heatmap` only works on point features lacking magnitudes. If you have gridded data with magnitudes, as in the cases you mentioned, this API won't work for you.

Heatmaps based on gridded magnitude data is a generalization of heatmaps based on point data, and I'm afraid there's quite a lot of math you or I would need to chew down to work an algorithm out. I don't personally know of any to achieve this currently in Python.

I'd recommend creates a grid of squares around each observation, and plotting that as a choropleth. You can think of this as a pixelated heatmap.

**Comment by stevencollinscn** [comment]
@ResidentMario Indeed...I'm looking at this [API](https://matplotlib.org/gallery/images_contours_and_fields/image_demo.html#interpolating-images) from `matplotlib` and see if I can figure it out. Sadly, this heatmap thing is the only part that is missing from my complete python workflow. Thank you for your kind reply.

**Comment by stevencollinscn** [comment]
@ResidentMario  I just saw a  [link](http://nbviewer.jupyter.org/gist/perrygeo/c426355e40037c452434) at the bottom of this [page](http://geopandas.org/mapping.html#other-resources) in the `GeoPandas` doc.

**Comment by ResidentMario** [comment]
Very cool! Maybe an implementation is easier than I thought.

**Comment by stevencollinscn** [comment]
@ResidentMario It would be great if there is a Python package that uses `scipy` [interpolation](https://docs.scipy.org/doc/scipy/reference/tutorial/interpolate.html#two-dimensional-spline-representation-procedural-bisplrep) module but is designed for geospatial point pattern analysis specifically like [IDW ](https://en.wikipedia.org/wiki/Inverse_distance_weighting)and other more advanced methods.

**Comment by StevenLi-DS** [comment]
@ResidentMario Here are some good resources.

https://mathlab.github.io/PyGeM/idw.html

https://stackoverflow.com/questions/3104781/inverse-distance-weighted-idw-interpolation-with-python

https://github.com/paulbrodersen/inverse_distance_weighting

**Comment by ResidentMario** [comment]
This is known as an [isopleth](https://en.wikipedia.org/wiki/Contour_line#Isopleths) in the literature.

**Comment by StevenLi-DS** [comment]
@ResidentMario The isopleth might be very useful but might not be what I was suggesting for since there are many methods for interpolation.

I just read a [book](https://www.academia.edu/37886944/GIS_Algorithms_-_Ningchuan_Xiao.pdf) this week and the author talked about a few different types of interpolation (e.g., IDW) often used in GIS in chapter 8 and shared his code [here](https://github.com/gisalgs/interpolation).

**Comment by ResidentMario** [comment]
Well, this feature request is for a plot function that takes point data with magnitudes as input, applies some sort of smoothing/interpolation function over it, and outputs a heatmap (which is an isopleth, in this context).

Interpolation can be tunable using a plot parameter. `gplt.quadtree` already defines an `agg` parameter for an aggregation function; an `isopleth` function could have a similar special parameter for a smoothing function whose default argument is an `idw` algorithm.

**Comment by StevenLi-DS** [comment]
as an arg? man... I think I would like such an API design and am definitely looking forward to it!
